### ISSUE TITLE
gtest compiled

### ISSUE THREAD

**Issue by oliviermattelaer** [madgraph5/madgraph4gpu#878]
I guess this is due to "my new way" to link the two repo.

But when i do create the code it now crashes with

```
	       18 | #include <gtest/gtest.h>
	          |          ^~~~~~~~~~~~~~~
	    compilation terminated.
	    In file included from runTest.cc:12:
	    MadgraphTest.h:13:10: fatal error: gtest/gtest.h: No such file or directory
	       13 | #include <gtest/gtest.h>
	          |          ^~~~~~~~~~~~~~~
	    compilation terminated.
	    make[2]: *** [runTest_cpp.o] Error 1
	    make[2]: *** Waiting for unfinished jobs....
	    make[2]: *** [testxxx_cpp.o] Error 1
	    c++ -shared -o /Users/omattelaer/Documents/git_workspace/3.1.1_lo_vectorization/PROC_TEST/lib/libmg5amc_common_cpp.so ./read_slha_cpp.o ./Parameters_sm_cpp.o
	    make[1]: *** [.cudacpplibs] Error 2
	    make: *** [madevent_cpp_link] Error 2

	Please try to fix this compilations issue and retry.
	Help might be found at https://answers.launchpad.net/mg5amcnlo.
	If you think that this is a bug, you can report this at https://bugs.launchpad.net/mg5amcnlo
```

I can (and will) revert to the previous method to link the two repo (which avoid that issue).
But we should we not avoid such compilation for madevent code?
(one of the point of CMS yesterday was actually that the compilation was too slow and this is likely a way to reduce that)

**Comment by valassi** [comment]
Hi Olivier, I will look at that.

Is this only on one special branch or everywhere?

Is there any special way to trigger this or a generate_events or launch will do?

And is there a way to switch on debugging, for instance to see the full log of the build?

Thanks
Andrea

**Comment by oliviermattelaer** [comment]
Hi Andrea,

Thanks for looking at this,

Sorry to not have replied earlier, looks like I was disconnected from my email server (at least I had a very quite period to relax). I did switch between various branches, so this is not branch specific.

As said, I guess I trigger this because I have now created a symlink in my home directory to avoid to have to change the symlink in the PLUGIN directory. Because of that it tries to compile the google test that it was ignoring before.

Maybe related, I try to run your tmad script and it also crash related to the google test:
Do you think it is related ? If not I can create another PR.
```
-- The C compiler identification is AppleClang 14.0.0.14000029
-- The C compiler identification is AppleClang 14.0.0.14000029
CMake Error at /Applications/CMake.app/Contents/share/cmake-3.24/Modules/CMakeDetermineCCompiler.cmake:226 (configure_file):
  No such file or directory
Call Stack (most recent call first):
  CMakeLists.txt:10 (project)


-- The CXX compiler identification is AppleClang 14.0.0.14000029
-- The CXX compiler identification is AppleClang 14.0.0.14000029
CMake Error at /Applications/CMake.app/Contents/share/cmake-3.24/Modules/CMakeDetermineCXXCompiler.cmake:226 (configure_file):
  No such file or directory
Call Stack (most recent call first):
  CMakeLists.txt:10 (project)


-- Detecting C compiler ABI info
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
CMake Error: The source directory "/Users/omattelaer/Documents/git_workspace/madgraph4gpu/test/googletest/build_unknown/CMakeFiles/CMakeTmp" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
CMake Error at /Applications/CMake.app/Contents/share/cmake-3.24/Modules/CMakeTestCCompiler.cmake:56 (try_compile):
  Failed to configure test project build system.
Call Stack (most recent call first):
  CMakeLists.txt:10 (project)


-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
-- Configuring incomplete, errors occurred!
See also "/Users/omattelaer/Documents/git_workspace/madgraph4gpu/test/googletest/build_unknown/CMakeFiles/CMakeOutput.log".
See also "/Users/omattelaer/Documents/git_workspace/madgraph4gpu/test/googletest/build_unknown/CMakeFiles/CMakeError.log".
make[3]: *** [googletest/build_unknown/Makefile] Error 1
make[2]: *** [../../../../../test/googletest/install_unknown/lib64//libgtest.a] Error 2
make[2]: *** Waiting for unfinished jobs....
c++  -I. -I../../src -O3  -std=c++17 -Wall -Wshadow -Wextra -ffast-math   -D__SSE4_2__  -DMGONGPU_FPTYPE_DOUBLE -DMGONGPU_FPTYPE2_DOUBLE -fPIC -c CPPProcess.cc -o build.sse4_d_inl0_hrd0/CPPProcess_cpp.o
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error at /Applications/CMake.app/Contents/share/cmake-3.24/Modules/CMakeTestCCompiler.cmake:56 (try_compile):
  Failed to configure test project build system.
Call Stack (most recent call first):
  CMakeLists.txt:10 (project)


-- Configuring incomplete, errors occurred!
See also "/Users/omattelaer/Documents/git_workspace/madgraph4gpu/test/googletest/build_unknown/CMakeFiles/CMakeOutput.log".
See also "/Users/omattelaer/Documents/git_workspace/madgraph4gpu/test/googletest/build_unknown/CMakeFiles/CMakeError.log".
make[3]: *** [googletest/build_unknown/Makefile] Error 1
make[2]: *** [../../../../../test/googletest/install_unknown/lib64//libgtest.a] Error 2
make[2]: *** Waiting for unfinished jobs....
c++  -I. -O3  -std=c++17 -Wall -Wshadow -Wextra -ffast-math   -D__SSE4_2__  -DMGONGPU_FPTYPE_DOUBLE -DMGONGPU_FPTYPE2_DOUBLE -fPIC -c Parameters_sm.cc -o build.sse4_d_inl0_hrd0/Parameters_sm_cpp.o
c++  -I. -O3  -std=c++17 -Wall -Wshadow -Wextra -ffast-math    -DMGONGPU_FPTYPE_DOUBLE -DMGONGPU_FPTYPE2_DOUBLE -fPIC -c Parameters_sm.cc -o build.none_d_inl0_hrd0/Parameters_sm_cpp.o
c++ -shared -o /Users/omattelaer/Documents/git_workspace/madgraph4gpu/epochX/cudacpp/gg_ttgg.mad/lib/build.sse4_d_inl0_hrd0/libmg5amc_common_cpp.so build.sse4_d_inl0_hrd0/read_slha_cpp.o build.sse4_d_inl0_hrd0/Parameters_sm_cpp.o
c++ -shared -o /Users/omattelaer/Documents/git_workspace/madgraph4gpu/epochX/cudacpp/gg_ttgg.mad/lib/build.none_d_inl0_hrd0/libmg5amc_common_cpp.so build.none_d_inl0_hrd0/read_slha_cpp.o build.none_d_inl0_hrd0/Parameters_sm_cpp.o
make[1]: *** [build.none_d_inl0_hrd0/.cudacpplibs] Error 2
make: *** [bldnone] Error 2
make[1]: *** [build.sse4_d_inl0_hrd0/.cudacpplibs] Error 2
make: *** [bldsse4] Error 2
```

**Comment by valassi** [comment]
Thanks Olivier fo rreporting, indeed it is better to avoid building tests in user code. This is now fixed in PR #879 that I will merge shortly. Closing
